python/src/iceberg/types.py

```diff
@@ -57,30 +57,65 @@ def __str__(self):
 
     @property
     def is_primitive(self) -> bool:
-        return self._is_primitive
+        return isinstance(self, PrimitiveType)
+
+
+class PrimitiveType(IcebergType):
+    """Base class for all Iceberg Primitive Types"""
+
+    _instances = {}  # type: ignore
+
+    def __new__(cls):
+        cls._instances[cls] = cls._instances.get(cls) or object.__new__(cls)
+        return cls._instances[cls]
+
+
+class FixedType(PrimitiveType):
+    """A fixed data type in Iceberg.
+
+    Example:
+        >>> FixedType(8)
+        FixedType(length=8)
+        >>> FixedType(8)==FixedType(8)
+        True
+    """
+
+    _instances: Dict[int, "FixedType"] = {}
 
+    def __new__(cls, length: int):
+        cls._instances[length] = cls._instances.get(length) or object.__new__(cls)
```

Comment:
Doesn't this need to pass `length` into `__new__` so that `__init__` is called correctly?